kafka-transport/src/main/java/com/esri/geoevent/transport/kafka/KafkaInboundTransport.java

```diff
@@ -1,157 +1,96 @@
-  private ConsumerConfig consumerConfig;
-  private String zkConnect;
-  private int numThreads;
-  private String topic;
-  private String groupId;
-
-  KafkaInboundTransport(TransportDefinition definition) throws ComponentException {
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.IntStream;
+
+class KafkaInboundTransport extends InboundTransportBase
+{
+  private static final BundleLogger             LOGGER                            = BundleLoggerFactory.getLogger(KafkaInboundTransport.class);
+  public static final  String                   GEOEVENT_TRANSPORT_CONSUMER_GROUP = "geoevent-transport-consumer-group";
+  private              String                   bootStrapServers;
+  private              int                      numThreads;
+  private              String                   topic;
+  private              Properties               configProperties;
+  private              AtomicBoolean            shutdownFlag                      = new AtomicBoolean(false);
+  private              ExecutorService          executorService;
+  private              List<KafkaEventConsumer> consumerList                      = new ArrayList<>();
+
+  KafkaInboundTransport(TransportDefinition definition) throws ComponentException
+  {
     super(definition);
   }
 
-  public boolean isClusterable() {
+  public boolean isClusterable()
+  {
     return true;
   }
 
   @Override
-  public void run()
+  public void afterPropertiesSet()
   {
-    setErrorMessage("");
-    setRunningState(RunningState.STARTED);
-    while (isRunning())
-    {
-      try
-      {
-        byte[] bytes = consumer.receive();
-        if (bytes != null && bytes.length > 0) {
-          ByteBuffer bb = ByteBuffer.allocate(bytes.length);
-          bb.put(bytes);
-          bb.flip();
-          byteListener.receive(bb, "");
-          bb.clear();
-        }
-      }
-      catch (MessagingException e)
-      {
-        LOGGER.error("", e);
-      }
-    }
-  }
-
-  @Override
-  public void afterPropertiesSet() {
-    zkConnect = getProperty("zkConnect").getValueAsString();
-    numThreads = Converter.convertToInteger(getProperty("numThreads").getValueAsString(), 1);
-    topic = getProperty("topic").getValueAsString();
-    groupId = getProperty("groupId").getValueAsString();
     super.afterPropertiesSet();
+    bootStrapServers = getProperty(KafkaInboundTransportDefinition.BOOTSTRAP_SERVERS).getValueAsString();
+    numThreads = Converter.convertToInteger(getProperty(KafkaInboundTransportDefinition.NUM_THREADS).getValueAsString(), 1);
+    topic = getProperty(KafkaInboundTransportDefinition.TOPIC).getValueAsString();
   }
 
   @Override
-  public void validate() throws ValidationException {
+  public void validate() throws ValidationException
+  {
     super.validate();
-    if (zkConnect.isEmpty())
+
+    if (bootStrapServers.isEmpty())
       throw new ValidationException(LOGGER.translate("ZKCONNECT_VALIDATE_ERROR"));
```

Comment:
Is this the right exception string? 
```suggestion
      throw new ValidationException(LOGGER.translate("ZKCONNECT_VALIDATE_ERROR"));
```

Reply:
This has been addressed